lib/charliecloud.py

```diff
@@ -1480,6 +1485,33 @@ def root_env():
       except KeyError:
          return None
 
+   def init(self):
+      """Ensure the storage directory exists, contains all the appropriate
+         top-level directories & metadata, and is the appropriate version."""
+      if (not os.path.isdir(self.root)):
+         op = "initializing"
+         v_found = None
+      else:
+         op = "upgrading"
+         if (not self.valid_p):
+            FATAL("storage directory seems invalid: %s" % self.root)
+         if (os.path.isfile(self.version_file)):
+            v_found = int(file_read_all(self.version_file))
+         else:
+            v_found = 1
+      if (v_found == STORAGE_VERSION):
+         VERBOSE("found storage dir v%d: %s" % (STORAGE_VERSION, self.root))
+      elif (v_found in {None, 1}):  # initialize/upgrade
+         INFO("%s storage directory: v%d %s" % (op, STORAGE_VERSION, self.root))
+         mkdirs(self.root)  # FIXME: use mkdir() when don't need two levels
```

Comment:
Can we not make this `mkdirs()` `mkdir()` now?

Reply:
No — because the default is `/var/tmp/$USER/ch-image` it will usually have to create both `$USER` and `ch-image`. Though it won't actually help if we change it to `/var/tmp/$USER.ch/ch-image`.

Reply:
See #1146 though.